File: Код/main4.py
from math import factorial


def main():
    x = float(input("Enter x: "))
    n = int(input("Enter n: "))

    print(sum([func(x, i) for i in range(1, n + 1)]))


def func(x, n):
    return x ** (n - 1) / factorial(n - 1)


if __name__ == '__main__':
    main()

File: Код/Main4.java
import java.util.Scanner;

public class Main4 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter x: ");
        double x = scanner.nextDouble();

        System.out.print("Enter n: ");
        int n = scanner.nextInt();

        scanner.close();

        double sum = 0;

        for (int i = 1; i <= n; i++) {
            sum += function(x, i);
        }

        System.out.println(sum);
    }

    private static double function(double x, int n) {
        return Math.pow(x, n - 1) / fact(n - 1);
    }

    private static long fact(int n) {
        return n == 0 ? 1 : (n * fact(n - 1));
    }
}
